nixos/modules/virtualisation/proxmox-image.nix

```diff
@@ -129,19 +129,12 @@ with lib;
         # Build qemu with PVE's patch that adds support for the VMA format
         vma = pkgs.qemu_kvm.overrideAttrs ( super: rec {
 
-          # proxmox's VMA patch doesn't work with qemu 7.0 yet
-          version = "6.2.0";
-          src = pkgs.fetchurl {
-            url= "https://download.qemu.org/qemu-${version}.tar.xz";
-            hash = "sha256-aOFdjkWsVjJuC5pK+otJo9/oq6NIgiHQmMhGmLymW0U=";
-          };
-
           patches = let
-            rev = "b37b17c286da3d32945fbee8ee4fd97a418a50db";
+            rev = "1976ca460796f28447b41e3618e5c1e234035dd5";
             path = "debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch";
             vma-patch = pkgs.fetchpatch {
-              url = "https://git.proxmox.com/?p=pve-qemu.git;a=blob_plain;h=${rev};f=${path}";
-              hash = "sha256-siuDWDUnM9Zq0/L2Faww3ELAOUHhVIHu5RAQn6L4Atc=";
```

Comment:
We could inline this entire let in.

Reply:
I've left the rev and path in the let, because those are the bits that will most often need to be changed. The rest has been inlined.

Reply:
The hash will also need to be changed if any of those are changed.

(Also, what does the change in the use of query parameter `h` to `hb` do?)

Reply:
Huh... You're right. I wasn't able to put my finger on why this felt cleaner before, but I think I see it now. If it was fully inlined, a change to the rev would change the entire url line. I wanted to keep the diff for a version bump as clean as possible.

The version I just pushed drops the top level let and adds it to url instead.

Reply:
As for the distinction between h and hb, (I don't know much about gitweb or the inner workings of git, but as far as I can tell...) "hb" is the commit ID, and "h" is the "tree ID". As far as I can tell, that hash is meant to uniquely identify a file in one state across commits.

```
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ git checkout dc9827a6a46d17fa0f0e5fb192defe0984881b9c
Previous HEAD position was 664ecf5 bump version to 6.2.0-4
HEAD is now at dc9827a update submodule and patches to 7.0.0
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ md5sum debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
a2b3f58995a86fe40a76ed84b86e5aca  debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ git ls-tree -r HEAD | grep debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
100644 blob ced8e74af871e15ab84da7fdf2ab067d27e0a2a2    debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ git checkout dc9827a6a46d17fa0f0e5fb192defe0984881b9c~1
Previous HEAD position was dc9827a update submodule and patches to 7.0.0
HEAD is now at 4e4b9ab bump version to 6.2.0-11
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ md5sum debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
2be50f0a1bf0f2d9ba4ea84f85404c0c  debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ git ls-tree -r HEAD | grep debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
100644 blob b37b17c286da3d32945fbee8ee4fd97a418a50db    debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ git checkout dc9827a6a46d17fa0f0e5fb192defe0984881b9c~2
Previous HEAD position was 4e4b9ab bump version to 6.2.0-11
HEAD is now at 39e84ba vma/alloc-track improvements
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ md5sum debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
2be50f0a1bf0f2d9ba4ea84f85404c0c  debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
[illustris@illustris-thinkpad:~/src/mnet/proxmox/pve-qemu]$ git ls-tree -r HEAD | grep debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
100644 blob b37b17c286da3d32945fbee8ee4fd97a418a50db    debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch
```

As you can see in the example above, any commit that changes the contents (or the attributres) of the file also change the tree hash of the file.

I changed it to hb in this commit because it makes more sense to refer to the file using the commit hash rather than the tree hash.